### PR TITLE
Update data_dictionary

### PR DESCRIPTION
**Pull request by SonQBChau**
if you want to add anything else, you can leave the comment

### CODE REVIEW

**Comment by harryyikhchan** [comment]
Just a little comment. Do we also need to add the label isNullAllowed? Some columns may allow blank fields so we make sure we don't impute those columns when we wrangle the data.

File: reference_material/data_dictionary [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+## api:
+The name of the API, e.g "Datamuse Dictionary", "DepartmentAPI"...
+Type: string
+
+## category
+The category of the API, e.g "Research & Education", "AI & Data Science"...
+Type: string
+
+## usage_base:
```
Comment (Jacq4nn) [inline reference_material/data_dictionary:26]: Perhaps we can add:
- Free: there are free limits, such as the number of calls that can be made for free. These limits are also available to hackers
- - Commercial: need registration. Additional layer of security

File: reference_material/data_dictionary [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+## sample_response:
+A sample response from the API.
+Type: JSON
+
+## tagset:
+List of keys in the sample response.
+Type: list
+
+## authentication:
```
Comment (Jacq4nn) [inline reference_material/data_dictionary:38]: If we are using an ordinal scale,

"None": not safe at all
"Path": risky
"others not listed": okay
"OAuth": strongest

File: reference_material/data_dictionary [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+## server_location:
+The location of the API, e.g "Australia", "United States"...
+Type: string
+
+## hosting_isp:
+The hosting ISP of the API, e.g "Amazon", "Cloudflare", "Google"...
+Type: string
+
+## server_name:
```
Comment (Jacq4nn) [inline reference_material/data_dictionary:58]: If the server name is exposed, there is a risk. Otherwise, low risk

**Review by Jacq4nn — COMMENTED** [review]
Hi Son

Included my comments based on what Baljeet mentioned yesterday in the meeting

**Comment by SonQBChau** [comment]
> Just a little comment. Do we also need to add the label isNullAllowed? Some columns may allow blank fields so we make sure we don't impute those columns when we wrangle the data.

Technically, these are not database, so any field can be null or considered bad data.

**Comment by SonQBChau** [comment]
depend, but ya, I just remove it

File: reference_material/data_dictionary [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+## tagset:
+List of keys in the sample response.
+Type: list
+
+## authentication:
+Authentication method, e.g "None", "OAuth 2.0", "Basic Authentication"...
+Type: string
+
+## security_test_category:
```
[the remark this answers is not in the document]
Comment (SonQBChau) [reply reference_material/data_dictionary:42]: This is subjective and we should base on https://owasp.org/www-project-api-security/

File: reference_material/data_dictionary (hunk 3 above)
Comment (SonQBChau) [reply reference_material/data_dictionary:58]: I agree, but any of the fields exposed are risks, not just the server name. They called it vector attack

**Comment by SonQBChau** [comment]
@Jacq4nn I added some of your comments. This is a living doc, so we can keep modifying.

**Pull request merged**